ydb/docs/en/core/concepts/glossary.md

```diff
@@ -0,0 +1,517 @@
+
+#### Prepare stage {#prepare-stage}
+
+The **prepare stage** is the transaction phase, during which the transaction body is registered on all participating shards.
+
+#### Execute stage {#execute-stage}
+
+The **execute stage** is the transaction phase in which the scheduled transaction is executed and the response is generated.
+
+In some cases, instead of [prepare](#prepare-stage) and execute, the transaction is immediately executed, and a response is generated.  For example, this happens for transactions involving only one shard or consistent reads from a snapshot.
+
+#### Dirty operations {#dirty-operations}
+
+In the case of read-only transactions, similar to "read uncommitted" in other database management systems, it might be necessary to read data that has not yet been committed to disk. This is called **dirty operations**.
+
+#### Read-write set {#rw-set}
+
+The **read-write set** or **RW set** is a set of shards that will participate in executing a [distributed transaction](#distributed-transactions). It combines read-set shards, from which data will be read, and write-set shards, in which modifications will be carried out.
+
+#### Read set {#read-set}
+
+The **read set** or **ReadSet data** is what participating shards forward during the transaction execution. In the case of data transactions, it may contain information about the state of [optimistic locks](#optimistic-locking), the readiness of the shard for commit, or the decision to cancel the transaction.
+
+#### Transaction proxy {#transaction-proxy}
+
+The **transaction proxy** or `TX_PROXY` is a service that orchestrates the execution of many [distributed transactions](#distributed-transactions): sequential phases, phase execution, planning, and aggregation of results. In the case of direct orchestration by other actors (for example, KQP data transactions), it is used for caching and allocation of unique [TxIDs](#txid).
+
+#### Transaction flags {#txflags}
+
+**Transaction flags** or **TxFlags** is a bitmask of flags that modify the execution of a transaction in some way.
+
+#### Transaction ID {#txid}
+
+**Transaction ID** or **TxID** is a unique identifier assigned to each transaction when it is accepted by {{ ydb-short-name }}.
+
+#### Transaction order ID {#transaction-order-id}
+
+A **transaction order ID** is a unique identifier assigned to each transaction during planning.
+
+#### PlanStep {#planstep}
+
+**PlanStep** or **step** is the logical time for which a set of transactions is planned to be executed.
+
+#### Mediator time {#mediator-time}
+
+**Mediator time** is the logical time before which (inclusive) the shard participant must know the entire execution plan. It is used to advance the time in the absence of transactions on a particular shard to determine whether it can read from a snapshot.
```

Comment:
надо написать, что это относится к механизму distr query execution

Reply:
```suggestion
During the distributed query execution, **mediator time** is the logical time before which (inclusive) the shard participant must know the entire execution plan. It is used to advance the time in the absence of transactions on a particular shard, to determine whether it can read from a snapshot.
```